tests/unit/test_models.py

```diff
@@ -140,12 +140,20 @@ def test_predict_without_allow_infeasible_solutions(self, model):
 
     @pytest.mark.unit
     def test_build(self, model):
-        model.build()
-        assert model.built_model is not None
+        if isinstance(model, pybop.lithium_ion.SPMe):
+            model.build(init_soc=1.0)
 
-        # Test that the model can be built again
-        model.build()
-        assert model.built_model is not None
+            # Test attributes with init_soc
+            assert model.built_model is not None
+            assert model.disc is not None
+            assert model.built_initial_soc is not None
+        else:
+            model.build()
+            assert model.built_model is not None
+
+            # Test that the model can be built again
```

Comment:
Let's open an issue to possibly merge `build` and `rebuild`, or at least ensure consistency between when the already-built model is returned and when it is rebuilt.